### DISCUSSION TITLE
Help needed to unbrick YI Dome U Pro 2K

### DISCUSSION THREAD

**Discussion by kyutov** [roleoroleo/yi-hack-Allwinner-v2#254] [Q&A] [answered]
Hi all,

I have bricked my cam YI Dome U Pro 2K. I have made a backup of the cam, but I cannot find the 3 uart pads (TX, RX and GND) on the mainboard in order to perform the unbrick process. Does anyone know where those pads are on this camera model? If needed, I can provide a photo of the mainboard.

**Comment by ghost** [comment]
Pictures might help. Maybe it's like this one? https://github.com/roleoroleo/yi-hack-MStar/issues/262

**Comment by kyutov** [comment]
@0-Al , unfortunately the mainboard in my cam is completely different. 

Here are the photos of the board in my cam:

![7E72801B-87D3-43CC-A7F5-6948F84C3107](https://user-images.githubusercontent.com/2218452/132947658-3a70b729-2925-43bc-a94e-9e16e5f7a429.jpeg)

![50EA2D07-7546-48C4-B1EF-5C61AB2FEB0B](https://user-images.githubusercontent.com/2218452/132947667-4ba5b904-16a1-48e7-83ad-22e0ead51b6c.jpeg)

I don’t see any TX, RX and GND labels on it.

**Reply by roleoroleo** [reply]
No labels on this pcb. Probably you need to test it manually.
Use the metallic ring under the screw as a GND.
And try to find the TX (cam side) connecting the RX (PC side) to a pin without solder it.
- Touch a pad with the pin.
- Switch on the cam
- Check if your putty shows text
- Change pad and restart the procedure

I would start with these:
![immagine](https://user-images.githubusercontent.com/39277388/132951460-4ee8be7d-bf26-43b2-9015-62eee247a5d1.png)
then
![immagine](https://user-images.githubusercontent.com/39277388/132951469-7e3b0485-c904-463c-8968-7a25bafe091d.png)
and
![immagine](https://user-images.githubusercontent.com/39277388/132951471-902b27ed-5b8f-46ef-9951-7b31b4bf5d6c.png)

**Reply by kyutov** [reply]
@roleoroleo , thank you for your answer. 
Ok, with the above procedure, I will try to find the couple TX (cam side) - RX (PC side), but after that how can I find the other couple RX (cam side) - TX (PC side)? By sending commands in putty and using the same procedure for the other couple TX-RX?

**Reply by roleoroleo** [reply]
Yes.

**Reply by kyutov** [reply]
Ok, clear. I will try to find the correct pads. I will report back once I mange to test. Thanks for now.

**Reply by kyutov** [reply]
@roleoroleo , I have managed to find the TX (cam side), but I'm struggling to find the RX (cam side). Actually I have managed to find two TX pads on the cam, here are they:

![132947667-4ba5b904-16a1-48e7-83ad-22e0ead51b6c](https://user-images.githubusercontent.com/2218452/134821173-a1c16ecb-d6a4-4e87-afb7-54005e8c2f01.jpeg)

When I touch one of those two pads, I'm able to see text in Putty. I have tried to find the RX pad (cam side) through all the remaining pads on the board, but no matter which pad I touch with the pin, I'm not able to send any commands to the cam.

Here is the log, which I'm seeing when I connect to the TX pad (cam side):


`[38]HELLO! BOOT0 is starting Nov 14 2019 20:01:31!
[43]BOOT0 commit : 81c18ee
[46]board init start
[48]set pll start
[50]set pll end
[52][pmu]: bus read error
[54]board init ok
[56]chip id check OK
[58]DRAM BOOT DRIVE INFO: V0.41
[61]DRAM CLK = 528 MHz
[63]DRAM Type = 2 (2:DDR2,3:DDR3)
[67]DRAMC read ODT  off.
[69]DRAM ODT off.
[71]DRAM SIZE =64 M
[78]DRAM simple test OK.
[80]rtc standby flag is 0x0, super standby flag is 0x0
[86]dram size =64
[88]key press : [
[90]spinor id is: c8 40 17, read cmd: 03
[94]Succeed in reading toc file head.
[98]The size of toc is 4c000.
[150]Entry_name        = optee
[153]Entry_name        = u-boot
[158]Entry_name        = dtb
[161]Jump to secend Boot.
MESSAGE: [0x0] TEE-CORE: OP-TEE version: sun8iw19p1_v0.6.0-12-g97f2688 #1 2019年 11月 08日 星期五 08:26:51 UTC arm
ERROR:   [0x0] TEE-CORE:platform_standby_fdt_parse:126: no pmu node
ERROR:   [0x0] TEE-CORE:sunxi_twi_parse_from_dt:84: no pmu node


U-Boot 2018.05 (Mar 08 2021 - 23:23:22 +0800) Allwinner Technology

[00.205]DRAM:  64 MiB
[00.208]Relocation Offset is: f9f94000
[00.219]secure enable bit: 0
[00.221]CPU=816 MHz,PLL6=600 Mhz,AHB=200 Mhz, APB1=100Mhz  MBus=132Mhz
[00.228]gic: sec monitor mode
[00.230]flash init start
[00.233]workmode = 0,storage type = 3
SF: Detected gd25q64 with page size 256 Bytes, erase size 4 KiB, total 8 MiB
[00.249]sunxi flash init ok
[00.252]Loading Environment from SUNXI_FLASH... start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x3e00, len: 0x100
OK
[00.313]try sprite_led_gpio config
xxxxxxxxxxxxxxlijun uboot xiaoyiledinit
[00.321]sprite_led_gpio start
[00.324](weak)update dtb dram start
[00.328]update dtb dram  end
[00.330]update dts
root_partition is rootfs
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
set root to /dev/mtdblock2
[00.393]update part info
[00.396]update bootcmd
mmc driver ver uboot2018:2019-8-15 16:34:00
get mem for descripter OK !
[00.408]get sdc_wipe fail.
[00.410]get sdc0 sdc_erase fail.
[00.413]get sdc0 sdc_boot fail.
[00.416]get sdc0 sdc_odly_50M fail.
[00.419]get sdc0 sdc_sdly_50M fail.
[00.423]get sdc0 sdc_odly_50M_ddr fail.
[00.426]get sdc0 sdc_sdly_50M_ddr fail.
[00.430]get sdc0 sdc_freq fail.
[00.432]get sdc0 sdc_b0p fail.
[00.435]get card0_boot_para:sdc_ex_dly_used fail
[00.440]get card-pwr-gpios handler:1119307360
[00.444]get card0_boot_para:time_pwroff:200ms
[00.450]Using default timing para
[00.653]init mmc 0 clock and io
devnum 0, prv 43fbd42c, bdesc 42b74844
SUNXI SD/MMC: 0
[00.660]==================== work mode: 0 0, sample_mode:0
[00.666]=============== start mmc_init_boot...
[00.684]Card did not respond to voltage select!
[00.688]mmc_init: -95, time 28
[00.691]mmc_init: mmc init fail, err -95
card0 init failed
No sd-card insert,will normal boot
Hit ctrl + q  to stop autoboot:  0
start: 0x2e0, len: 0x1
start: 0x2e1, len: 0x1
start: 0x2e2, len: 0x2
[01.714]partinfo: name boot, start 0x20, size 0xe80
start: 0x300, len: 0x40
start: 0x340, len: 0xe10
[02.190]android.hardware = sun8iw19p1
## Booting kernel from Legacy Image at 45000000 ...
   Image Name:   ARM OpenWrt Linux-4.9.118
   Image Type:   ARM Linux Kernel Image (uncompressed)
   Data Size:    1875896 Bytes = 1.8 MiB
   Load Address: 40008000
   Entry Point:  40008000
[02.257]Starting kernel ...


`

**Reply by roleoroleo** [reply]
>  I have managed to find the TX (cam side), but I'm struggling to find the RX (cam side). Actually I have managed to find two TX pads on the cam, here are they:

This is strange.
It would not be smart to put 2 identical pads on the pcb.
Check if there is a short circuit on your adapter: maybe you are seeing the echo of the tx pad.

**Reply by kyutov** [reply]
Ok, I will check. Do you have any recommendations or ideas how to find the RX pad (cam side)?

So far I tried to find the RX pad by touching it with the pin and during the boot of the cam clicking “Ctrl + q” in order to stop the autoboot, but I didn’t managed the boot so I suppose that I have not touched the appropriate RX pad.

**Reply by roleoroleo** [reply]
Usually the RX and TX pins are very close.

I think you can enter in u-boot mode pressing "Enter".
You should see the prompt "=>".

**Comment by ghost** [comment]
Here's a thought in case the serial path is not going well, you might be able to join the backup files to make the original 8MiB image and then use a cheap SPI programmer with SOP clip to flash it.

**Reply by ghost** [reply]
i m also interested in how to build the original 8MiB image. could you explain it? is it possible to flash it serial? my camera is also brcked after flashing the wrong yi hack image. 
i have serial connection but flashing mtb0004.bin doesn t fixe the problem

**Comment by roleoroleo** [comment]
This is the mtd table:
```
dev:    size   erasesize  name
mtd0: 00060000 00001000 "uboot"
mtd1: 001d0000 00001000 "boot"
mtd2: 00120000 00001000 "rootfs"
mtd3: 00340000 00001000 "home"
mtd4: 00130000 00001000 "backup"
mtd5: 00020000 00001000 "env"
mtd6: 00010000 00001000 "mfg"
mtd7: 00010000 00001000 "conf"
mtd8: 00000000 00000000 "UDISK"
```
Simply concatenate all mtd files and you will have the whole image.
About SOP clip, it depends on the pcb, sometimes it is necessary to unsolder the chip.

But, if the cam is ok (it's not broken) you must be able to unbrick following the unbrick wiki.
mtd2 and mt3 are read only and it's difficult to damage them.
So, mtd4 should be enough.

**Reply by ghost** [reply]
I went through the unbrick tutorial and at the end the cam does not boot. 
I don't know what I should do.

**Comment by ghost** [comment]
The conf section gets modified as well but could not say if it can cause bricking. 

I guess I'm used to using a programmer so I do it that way, been good to me for many years now and cost something like $5 with clip, 1.8V converter (for chips that are 1.8V otherwise it uses 3.3V) and a couple of SOIC sockets.

![300](https://user-images.githubusercontent.com/90365717/133037522-fe967e5e-3825-40c2-b725-68ba3f8737f5.jpg)

**Reply by roleoroleo** [reply]
> The conf section gets modified as well but could not say if it can cause bricking.

Yes, the conf section changes when you edit the configuration with the app.
If the cam crashes or restarts during a write, it could be a problem.
About the hack, it changes conf section only if you set the wifi from the web page.

**Comment by ngutman** [comment]
Jumping in on the conversation - I've used a logic analyzer to test the uart pads, I can definitely see uart communication but for some reason the decoder doesn't manage to correctly decode the stream.. I'm guessing it has something to do with the bits/parity configuration - anyone knows the standard for yi cameras? I tried using 8N1 (8 data bits, 1 parity, most-significant-bit)

This is how the signal on the RX pad looks like 
<img width="1919" alt="image" src="https://user-images.githubusercontent.com/1540134/135393313-f643719f-9916-41f8-a6cc-b45b2eca0fdb.png">

This is the uart configuration
<img width="376" alt="image" src="https://user-images.githubusercontent.com/1540134/135393383-1f844d31-5fd1-41bb-8687-fc93569c2203.png">

Tomorrow I'll get my cp210x usb cable but I'm just a bit confused 🤷

**Reply by roleoroleo** [reply]
My cam is 115200 8N1.

**Reply by ngutman** [reply]
Thanks, 8N1 worked!
Another issue was sampling rate (needed at least 4x the bandwidth, used 100Khz at first once I switched to 500Khz it worked)

**Comment by ghost** [comment, marked as answer]
@kyutov You have found the right pads, one is TX and the other is RX. I suspect you have used a non Yi adapter and USB data cable which is causing "the short?" because the camera RX and TX lines are routed to the USB connector at the bottom, no need to even take apart the camera to use the serial debug if willing to make a non standard USB cable connection. That is a cable that doesn't connect to the usual D+ D- connections from a USB host but splits of the two data wires for 232 use along with a piggyback to the ground wire. The 5V and ground still supply the camera. I can try to make some further details if the explanation is not too clear. Or you could try and find a USB charging cable that only uses two wires and stick with those two pesky testing pads.

**Reply by ngutman** [reply]
@0-Al You have saved my camera! 🥳 - after an unfortunate accident in which I ripped the TX uart pad from the PCB (they are *so* small to work with), I was sure I need to throw away the camera since I corrupted the backup partition and had to get serial connection.

Thanks to your tip I resurrected my camera! Here's a picture of the custom usb cable I made -

![image](https://user-images.githubusercontent.com/1540134/135741464-48bc3f99-153e-46fb-9c0d-a9cd8eb61c1f.png)

**Reply by ghost** [reply]
Cool. :+1:  Let's hope it's not too late for @kyutov :worried:

**Reply by kyutov** [reply]
@0-Al You are absolutely right! I have used a normal USB data cable to power my cam. I tried to power it with the Yi power cable and in that way I was able to communicate with the RX and TX pads and to successfully unbrick my cam  🥳
Thank you! 👍

@roleoroleo , I have marked the right pads on my board in case you want to add it to the unbrick guide.

![132947667-4ba5b904-16a1-48e7-83ad-22e0ead51b6c](https://user-images.githubusercontent.com/2218452/135745341-e80e0342-d990-4bdb-8afe-3fc605ba40f8.jpg)

Additionally the following command `sunxi_card0_probe` was not recognised by my cam, but even without it, I was able to access the SD card and complete the unbrick procedure. Probably the SD card is enabled by default for my model.

I will make a USB cable just like @ngutman did, but in my case, my USB uart adapter provides 5V lead as well, so I I'm planning to power the cam directly from that lead and I will connect a microUSB cable directly to my USB uart adapter. Once I manage to communicate with the cam through the USB port of my cam, I will try to hack it again. I hope that in case I brick the cam again, I will be able to see the reason that is causing the bricking. 
If I can help with logs or something else from my cam to make a stable version of the hack for that model (it is **h60ga**), let me know and I will assist.

**Reply by ghost** [reply]
Great news, sorry it took so long. 

By linking D+ and D- a typical 5V adapter identifies itself as a Dedicated Charging Port (DCP). This means with a normal USB data cable a 232 loopback can be introduced which might possibly be problematic to normal operation and maybe explains why the UART is set to silent after the short boot stage. I wonder if Yi-Hack sends data through the port? I haven't tried it with the Dome U pro.

Edit: Have to wonder if the Yi 1080p Home also connects the serial comm's to the USB connector given this conversation. [https://github.com/roleoroleo/yi-hack-Allwinner-v2/issues/188](url)
